craft_parts/plugins/python_v2/python_plugin.py

```diff
@@ -0,0 +1,115 @@
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Version 2 of the Python plugin."""
+
+import shlex
+from textwrap import dedent
+from typing import Literal
+
+from typing_extensions import override
+
+from craft_parts.plugins import Plugin, PluginProperties
+
+
+class PythonPluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the python plugin."""
+
+    plugin: Literal["python"] = "python"
+
+    python_requirements: list[str] = []
+    python_packages: list[str] = []
+
+    # part properties required by the plugin
+    source: str  # pyright: ignore[reportGeneralTypeIssues]
+
+
+class PythonPlugin(Plugin):
+    """Python plugin, version 2.
+
+    Unlike the original Python plugin, this one does *not* require the creation of
+    virtual environments to work. Instead, it uses pip's support for both a) installing
+    directly into the user's base dir and b) using a Python interpreter that is not the
+    one running pip initially. This allows lighter payloads as the -venv package is
+    not required, and interacts well with usrmerged directories.
+
+    The downside is that the payload generated by the plugin assumes that the primed
+    contents will include the Python interpreter; if this is not the case, the runtime
+    will need to add the payload to Python's path (e.g. via PYTHONPATH).
+    """
+
+    properties_class = PythonPluginProperties
+    _options: PythonPluginProperties
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        """Return a set of required packages to install in the build environment."""
+        return set()
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Return a dictionary with the environment to use in the build step."""
+        return {
+            "PIP_USER": "1",
+            "PYTHONUSERBASE": str(self._part_info.part_install_dir),
+            "PIP_BREAK_SYSTEM_PACKAGES": "1",
+            "PIP_PYTHON": "$(which python3)",
+        }
+
+    @override
+    def get_build_commands(self) -> list[str]:
+        """Return a list of commands to run during the build step."""
+        pip_calls: list[str] = []
+
+        # First the requirements
+        pip_calls.extend(
+            f"pip install -r {req}" for req in self._options.python_requirements
```

Comment:
Should we perhaps make something for getting `pip` not from the install dir? If an old version of `pip` is in the requirements, the second build could potentially fail.

Reply:
I understand the concern but I wonder about the best way to support this. This thought (bad pip) came to me during development but since I don't have a concrete case for it now I thought we might postpone the decision a bit. It's an advantage of testing this in Rockcraft's 25.10 support - we can start with a very trim plugin and add more features as users request them with clear use-cases.

How does that sound?

Reply:
We can handle that problem by later introducing an environment variable `$PIP_BIN` or similar that defaults to `/usr/bin/pip`